src/tqec/computation/block_graph.py

```diff
@@ -273,3 +273,30 @@ def shift_min_z_to_zero(self) -> BlockGraph:
                 pipe.kind,
             )
         return new_graph
+
+    def rotate(
+        self,
+        rotation_axis: Direction3D = Direction3D.Y,
+        num_90_degree_rotation: int = 1,
+        counterclockwise: bool = True,
+    ) -> BlockGraph:
+        """Rotate the graph around an axis by ``num_90_degree_rotation * 90`` degrees and
+        return a new rotated graph.
+
+        Args:
+            rotation_axis: The axis around which to rotate the graph.
+            num_90_degree_rotation: The number of 90-degree rotations to apply to the graph.
+            counterclockwise: Whether to rotate the graph counterclockwise. If set to False,
+                the graph will be rotated clockwise. Defaults to True.
+
+        Returns:
+            A data-independent copy of the graph rotated by the given number of 90-degree rotations.
+        """
+        n = num_90_degree_rotation % 4
+
+        if n == 0:
+            return self.copy()
+
+        zx = self.to_zx_graph()
+        rotated_zx = zx.rotate(rotation_axis, n, counterclockwise)
+        return rotated_zx.to_block_graph(self.name + " rotated")
```

Comment:
I like to be able to follow what transformations have been performed on something from its name. In this spirit, I would rather name the new graph using
```suggestion
        fname = "R" + str(n * 90 if not counterclockwise else (4 - n) * 90)
        return rotated_zx.to_block_graph(f"{fname}({self.name})")
```
This is very much a detail.